cppapi/server/device.h

```diff
@@ -454,6 +454,18 @@ class DeviceImpl : public virtual POA_Tango::Device
  */
 	virtual void always_executed_hook(void) {};
 
+/**
+ * Hook method.
+ *
+ * Default method to implement an action necessary on a device after server has been initialized.
+ *
+ * @exception DevFailed This method does not throw exception but a
+ * redefined method can.
+ * Click <a href="https://tango-controls.readthedocs.io/en/latest/development/advanced/IDL.html#exceptions">here</a> to read
+ * <b>DevFailed</b> exception specification
+ */
+    virtual void server_init_hook(const DServer &)
+    {};
```

Comment:
This Pull Request will result in a new major version of Tango because we lose the binary compatibility by adding this virtual function. This is not a problem but we should keep this in mind when updating the version number.
By the way, I created [this wiki page](https://github.com/tango-controls/cppTango/wiki/Policies-Binary-Compatibility-Issues) which refers to a web page explaining the Do's and Don'ts to preserve binary compatibility.